From 057696bbef9a0b48731edeac7425005a7e928706 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 14 Sep 2021 06:32:23 +0300
Subject: [PATCH] [SW-54874] gaudi2: arc: add macro for start address

In secureboot, ppboot and preboot start address are shifted by 0xC and
0x8 respectively to add size and version details at start of image.
This patch add macros to make code more readable.

Change-Id: I77b79659b20b752df6eda5a65fb48e9aae4ba84b
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index ec5c655..d7a4471 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -31,8 +31,10 @@
 #define ARC_ICCM_BASE		(GEN_REGION_BASE + 0x7040000)
 #define ARC0_ICCM_BASE		(GEN_REGION_BASE + 0x7000000)
 #define SP_SRAM_PREBOOT_OFFSET	0x1000
-#define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
-#define PREBOOT_TEXT_BASE	(SP_SRAM_BASE + SP_SRAM_PREBOOT_OFFSET + 0x8)
+#define PREBOOT_START_OFFSET	0x8
+#define PPBOOT_START_OFFSET	0xC
+#define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + PPBOOT_START_OFFSET)
+#define PREBOOT_TEXT_BASE	(SP_SRAM_BASE + SP_SRAM_PREBOOT_OFFSET + PREBOOT_START_OFFSET)
 
 #define SP_SRAM_BOOT_INFO_OFFSET	0x1800
 #define BOOT_INFO_BASE		(SP_SRAM_BASE + SP_SRAM_BOOT_INFO_OFFSET)
